newIDE/app/src/VariablesList/index.js

```diff
@@ -173,23 +194,45 @@ export default class VariablesList extends React.Component<Props, State> {
         variable={variable}
         disabled={depth !== 0}
         depth={depth}
+        type={type}
         errorText={
           this.state.nameErrors[variable.ptr]
             ? 'This name is already taken'
             : undefined
         }
         onChangeValue={text => {
-          variable.setString(text);
+          // if it's an object variable edited, create an instance variable in it's place marked as 'inherited'
```

Comment:
As this is becoming large, this should probably go in a method called `updateOrDefineVariable`. Naming the method is also a way to autodocument the code :)